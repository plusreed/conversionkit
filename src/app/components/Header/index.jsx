import React from 'react';

const Header = props => (
    <>
        <div className="h-auto p-2 bg-gray-900 pl-4">
            <h1 className="font-bold text-white text-xl">{props.text || "ConversionKit"}</h1>
        </div>
    </>
);

export default Header;