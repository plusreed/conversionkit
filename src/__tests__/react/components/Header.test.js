import React from 'react';
import ReactDOM from 'react-dom';
import Header from '../../../app/components/Header';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Header />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('contains text when text prop is passed', () => {
  const div = document.createElement('div')
  ReactDOM.render(<Header text={"hey"} />, div)
  expect(div.innerHTML).toMatch(/hey/)
})
