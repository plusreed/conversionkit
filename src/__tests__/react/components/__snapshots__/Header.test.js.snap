// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot when no text is provided 1`] = `
<div
  className="h-auto p-2 bg-gray-900 pl-4"
>
  <h1
    className="font-bold text-white text-xl"
  >
    ConversionKit
  </h1>
</div>
`;

exports[`matches snapshot when text is provided 1`] = `
<div
  className="h-auto p-2 bg-gray-900 pl-4"
>
  <h1
    className="font-bold text-white text-xl"
  >
    hey
  </h1>
</div>
`;
